### PR TITLE
feature/team_detail

### PR DESCRIPTION
**Pull request by nhanviet1**
## Related Tickets
- [#Task 57037: Complete the Team's Detail Screen](https://edu-redmine.sun-asterisk.vn/issues/57037)
## WHAT
- Complete the Team's Detail Screen
- Add the local history screen for saving drivers seen recently
## Evidence (Screenshot or Video)
image
image

## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://sal.vn/WtYPAy |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Redmine | Sun* Redmine working process  | https://sal.vn/piPoQn |<li>- [ ] yes</li>|<li>- [ ] yes</li>

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -15,6 +15,21 @@
         android:supportsRtl="true"
         android:theme="@style/Theme.Formula1"
         tools:targetApi="31">
+        <activity
+            android:name=".view.mainscreen.HistoryActivity"
+            android:exported="false">
```
Comment (hungvv-1301) [inline app/src/main/AndroidManifest.xml]: android:screenOrientation="portrait"

File: app/src/main/res/layout/activity_history.xml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:paddingTop="@dimen/dp_10"
+        android:paddingBottom="@dimen/dp_20"
+        app:layoutManager="androidx.recyclerview.widget.LinearLayoutManager"
+        app:layout_constraintTop_toBottomOf="@id/img_back"
+        tools:itemCount="4"
+        tools:listitem="@layout/item_driver_standing" />
+
+</androidx.constraintlayout.widget.ConstraintLayout>
```
Comment (hungld-1302) [inline app/src/main/res/layout/activity_history.xml]: endline

File: app/src/main/java/com/example/formula1/view/mainscreen/HistoryActivity.kt [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    private val driverAdapter = SeenDriverAdapter(::onItemClick)
+    private val viewModel: StandingViewModel by viewModel()
+    private val listTeamUnique = mutableListOf<TeamUnique>()
+    private var carURL: Int? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setup()
+        setupObservers()
+        viewModel.getDriverLocal()
```
Comment (hungld-1302) [inline app/src/main/java/com/example/formula1/view/mainscreen/HistoryActivity.kt:30]: cho vào init viewmodel

File: app/src/main/java/com/example/formula1/view/adapter/SeenDriverAdapter.kt [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        private val itemBinding: ItemDriverStandingBinding, onclickItem: (Driver) -> Unit
+    ) : BaseViewHolder<Driver, ItemDriverStandingBinding>(itemBinding, onclickItem) {
+        override fun onBindData(data: Driver) {
+            super.onBindData(data)
+            itemBinding.textPosition.text = data.number.toString()
+            itemBinding.textDriverName.text = data.name
+            itemBinding.textScore.gone()
+            itemBinding.textTeamName.text = data.nameTag
+            val context = itemBinding.root.context
+            context.loadCoverImage(data.image as String, itemBinding.imgDriver)
```
Comment (hungvv-1301) [inline app/src/main/java/com/example/formula1/view/adapter/SeenDriverAdapter.kt]: assert not null

**Review by nguyennhatminh230801 — APPROVED** [review]

File: app/src/main/java/com/example/formula1/data/repository/SearchRepositoryImplement.kt [hunk 5]
```diff
@@ -49,4 +50,23 @@ class SearchRepositoryImplement(
...
+    ): DataResult<TeamSearchResponse> {
+        return getResult(dispatcher) {
+            remote.getTeamSearchResult(id)
+        }
+    }
+
+    override suspend fun getSearchDriverByTeam(
+        teamID: Int,
+        season: String,
+        dispatcher: CoroutineDispatcher
```
Comment (vietnh-1109) [inline app/src/main/java/com/example/formula1/data/repository/SearchRepositoryImplement.kt]: truyền lại cho a nhé

**Pull request merged**